### PR TITLE
securityadmin: Replace TransportClient by RestHighLevelClient

### PR DESCRIPTION
**Pull request by rs-eliatra**
### Description
[Describe what this change achieves]
* Category = Maintenance
* Why these changes are required?
Check description in: https://github.com/opensearch-project/security/issues/1578

### Issues Resolved
https://github.com/opensearch-project/security/issues/1578

### Testing
Testing done: 
- unit testing
- manual tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java [hunk 1]
```diff
@@ -459,7 +461,8 @@ private boolean checkFilePermissions(final Path p) {
                 handlers.add(new DashboardsInfoAction(settings, restController, Objects.requireNonNull(evaluator), Objects.requireNonNull(threadPool)));
                 handlers.add(new TenantInfoAction(settings, restController, Objects.requireNonNull(evaluator), Objects.requireNonNull(threadPool),
 				Objects.requireNonNull(cs), Objects.requireNonNull(adminDns), Objects.requireNonNull(cr)));
-
+                handlers.add(new SecurityConfigUpdateAction(settings, restController,Objects.requireNonNull(threadPool), adminDns, configPath, principalExtractor));
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java:464]: Why do these handlers need to be added to replace TransportClient?

File: src/test/java/org/opensearch/security/InitializationIntegrationTests.java [hunk 2]
```diff
@@ -46,7 +46,6 @@
 import org.opensearch.common.transport.TransportAddress;
 import org.junit.Assert;
 import org.junit.Test;
-
```
Comment (cliu123) [inline src/test/java/org/opensearch/security/InitializationIntegrationTests.java:49]: Please remove this change if not needed.

File: src/main/java/org/opensearch/security/action/configupdate/ConfigUpdateNodeResponse.java [hunk 3]
```diff
@@ -78,4 +80,14 @@ public void writeTo(StreamOutput out) throws IOException {
     public String toString() {
         return "ConfigUpdateNodeResponse [updatedConfigTypes=" + Arrays.toString(updatedConfigTypes) + ", message=" + message + "]";
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/action/configupdate/ConfigUpdateNodeResponse.java:85]: Why is this needed?

File: src/main/java/org/opensearch/security/filter/SecurityRestFilter.java [hunk 4]
```diff
@@ -184,7 +186,9 @@ private boolean checkAndAuthenticateRequest(RestRequest request, RestChannel cha
 
         Matcher matcher = PATTERN_PATH_PREFIX.matcher(request.path());
         final String suffix = matcher.matches() ? matcher.group(2) : null;
-        if(request.method() != Method.OPTIONS && !(HEALTH_SUFFIX.equals(suffix))) {
+        if(request.method() != Method.OPTIONS
+                && !(HEALTH_SUFFIX.equals(suffix))
+                && !(WHO_AM_I_SUFFIX.equals(suffix))) {
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/filter/SecurityRestFilter.java:191]: Are these changes needed for replacing TransportClient?

File: src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java [hunk 5]
```diff
@@ -459,7 +461,8 @@ private boolean checkFilePermissions(final Path p) {
...
+                handlers.add(new SecurityWhoAmIAction(settings ,restController,Objects.requireNonNull(threadPool), adminDns, configPath, principalExtractor));
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java:465]: Why do these handlers need to be added to replace TransportClient?

File: src/main/java/org/opensearch/security/rest/SecurityConfigUpdateAction.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.opensearch.rest.RestRequest.Method.PUT;
+import static org.opensearch.security.dlic.rest.support.Utils.addRoutesPrefix;
+
+public class SecurityConfigUpdateAction extends BaseRestHandler {
+
+    private static final List<Route> routes = addRoutesPrefix(ImmutableList.of(
+            new Route(PUT, "/configupdate")),
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/rest/SecurityConfigUpdateAction.java:33]: Why is it necessary to add a new endpoint? Can't we use the existing one `/securityconfig` here?

File: src/main/java/org/opensearch/security/rest/SecurityWhoAmIAction.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+				}
+
+				channel.sendResponse(response);
+			}
+		};
+	}
+
+	@Override
+	public String getName() {
+		return "Search Guard Who am i";
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/rest/SecurityWhoAmIAction.java:118]: Please remove `Search Guard`

File: src/main/java/org/opensearch/security/rest/SecurityWhoAmIAction.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+
+import static org.opensearch.rest.RestRequest.Method.GET;
+import static org.opensearch.rest.RestRequest.Method.POST;
+import static org.opensearch.security.dlic.rest.support.Utils.addRoutesPrefix;
+
+
+public class SecurityWhoAmIAction extends BaseRestHandler {
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/rest/SecurityWhoAmIAction.java:38]: Why `SecurityWhoAmIAction` and `SecurityConfigUpdateAction` are needed for this PR?

File: src/main/java/org/opensearch/security/rest/SecurityWhoAmIAction.java (hunk 7 above)
Comment (rs-eliatra) [reply src/main/java/org/opensearch/security/rest/SecurityWhoAmIAction.java]: done

File: src/test/java/org/opensearch/security/InitializationIntegrationTests.java (hunk 2 above)
Comment (rs-eliatra) [reply src/test/java/org/opensearch/security/InitializationIntegrationTests.java]: done

File: src/main/java/org/opensearch/security/filter/SecurityRestFilter.java (hunk 4 above)
Comment (rs-eliatra) [reply src/main/java/org/opensearch/security/filter/SecurityRestFilter.java:191]: They are needed to keep SecurityAdmin to offer:
* whoami https://github.com/opensearch-project/security/blob/04d5dcb6dd6b93b1cfca55c6c41b889da9ccc1a3/src/main/java/org/opensearch/security/tools/SecurityAdmin.java#L244
* healthcheck https://github.com/opensearch-project/security/blob/04d5dcb6dd6b93b1cfca55c6c41b889da9ccc1a3/src/main/java/org/opensearch/security/tools/SecurityAdmin.java#L617

File: src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java (hunk 5 above)
Comment (rs-eliatra) [reply src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java:465]: https://github.com/opensearch-project/security/pull/1638#discussion_r813423437

File: src/main/java/org/opensearch/security/support/WildcardMatcher.java [hunk 9]
```diff
@@ -252,7 +252,7 @@ public boolean matchAny(Collection<String> candidates) {
         return matchAny(candidates.stream());
     }
 
-    public boolean matchAny(String[] candidates) {
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/support/WildcardMatcher.java:255]: This is being called in [here](https://github.com/opensearch-project/security/blob/12ae0cba4e5bdbe45e85f35078cc785c99cf9ab6/src/main/java/org/opensearch/security/transport/DefaultInterClusterRequestEvaluator.java#L99).

File: src/main/java/org/opensearch/security/rest/SecurityWhoAmIAction.java (hunk 8 above)
Comment (rs-eliatra) [reply src/main/java/org/opensearch/security/rest/SecurityWhoAmIAction.java:38]: https://github.com/opensearch-project/security/pull/1638#discussion_r813423437

File: src/main/java/org/opensearch/security/support/WildcardMatcher.java (hunk 9 above)
Comment (rs-eliatra) [reply src/main/java/org/opensearch/security/support/WildcardMatcher.java:255]: Does it break something?

mostly: 
```
matchAny(String[] candidates)
```
==
```
matchAny(String... candidates)
```

File: src/main/java/org/opensearch/security/tools/SecurityAdmin.java [hunk 10]
```diff
@@ -213,7 +217,6 @@ public static int execute(final String[] args) throws Exception {
         options.addOption(Option.builder("r").longOpt("retrieve").desc("retrieve current config").build());
         options.addOption(Option.builder("f").longOpt("file").hasArg().argName("file").desc("file").build());
         options.addOption(Option.builder("t").longOpt("type").hasArg().argName("file-type").desc("file-type").build());
-        options.addOption(Option.builder("tsalias").longOpt("truststore-alias").hasArg().argName("alias").desc("Truststore alias").build());
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/tools/SecurityAdmin.java:216]: Why removing this?

File: src/main/java/org/opensearch/security/tools/SecurityAdmin.java [hunk 11]
```diff
@@ -275,7 +278,6 @@ public static int execute(final String[] args) throws Exception {
         String type = null;
         boolean retrieve = false;
         String ksAlias = null;
-        String tsAlias = null;
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/tools/SecurityAdmin.java:278]: Why removing this?

File: src/main/java/org/opensearch/security/tools/SecurityAdmin.java [hunk 12]
```diff
@@ -548,30 +509,47 @@ public static int execute(final String[] args) throws Exception {
...
             }
-            
-            if(reload) { 
-                ConfigUpdateResponse res = tc.execute(ConfigUpdateAction.INSTANCE, new ConfigUpdateRequest(getTypes(false))).actionGet();                
-                if(res.hasFailures()) {
-                    System.out.println("ERR: Unabe to reload config due to "+res.failures());
+
+            if(reload) {
+                Response res = restHighLevelClient.getLowLevelClient()
+                        .performRequest(new Request("PUT", "/_plugins/_security/configupdate?config_types=" + Joiner.on(",").join(getTypes(false))));
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/tools/SecurityAdmin.java:543]: Same question as above: why can't it use  the existing endpoint `/securityconfig`?

File: src/main/java/org/opensearch/security/tools/SecurityAdmin.java [hunk 13]
```diff
@@ -1379,15 +1378,177 @@ private static boolean validateConfigFile(String file, CType cType, int version)
...
+		HttpHost httpHost = new HttpHost(hostname, port, "https");
+
+		RestClientBuilder restClientBuilder = RestClient.builder(httpHost)
+				.setHttpClientConfigCallback(
+						builder -> builder.setSSLStrategy(
+								new SSLIOSessionStrategy(
+										sslContext,
+										supportedProtocols,
+										supportedCipherSuites,
+										hnv
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/tools/SecurityAdmin.java:1415]: The format looks a bit off. Is it formatted by IDE?

File: src/test/java/org/opensearch/security/SecurityAdminTests.java [hunk 14]
```diff
@@ -59,15 +64,89 @@ public void testSecurityAdmin() throws Exception {
...
-        
-        Assert.assertEquals(HttpStatus.SC_OK, (res = rh.executeGetRequest("_opendistro/_security/health?pretty")).getStatusCode());
+        RestHelper rh = restHelper();
+
+        Assert.assertEquals(HttpStatus.SC_OK, (rh.executeGetRequest("_opendistro/_security/health?pretty")).getStatusCode());
     }
-    
+
+    @Test
+    public void testSgAdminInvalidCert() throws Exception {
```
Comment (cliu123) [inline src/test/java/org/opensearch/security/SecurityAdminTests.java:73]: Please rename the method name to `testSecurityAdminInvalidCert`

File: src/main/java/org/opensearch/security/rest/SecurityConfigUpdateAction.java (hunk 6 above)
Comment (nibix) [reply src/main/java/org/opensearch/security/rest/SecurityConfigUpdateAction.java:33]: This patch does not modify the basic working principle of securityadmin, it just moves the communication between securityyadmin and the backend from the transport protocol to the REST layer. 

`securityadmin` uses a number of different actions to work:

- `IndexAction` to update the security index. This is a standard OpenSearch action and is thus also available on the REST layer
- `ConfigUpdateAction` to notify the security plugin that the index contents have been changed and that the content needs to be reloaded. So far, it was only necessary as transport action, thus it was not exposed on the REST layer. Thus, we now need to expose it on the REST layer which is done by the class in question.
- `WhoAmIAction` to check authentication. This is also used by `securityadmin` for auxilliary tasks. This way also not exposed on REST before.

Using the existing `config` REST APIs won't work, as these only work on single config entries, not the whole config. Securityadmin on the other hand must be able to update the whole configuration of several config types at one time.

Of course, it would be nicer to implement all this using a brand-new dedicated REST API, and generally this would be the way I would recommend. However, this is not possible on the given time-frame.

File: src/main/java/org/opensearch/security/action/configupdate/ConfigUpdateNodeResponse.java (hunk 3 above)
Comment (nibix) [reply src/main/java/org/opensearch/security/action/configupdate/ConfigUpdateNodeResponse.java:85]: For exposing Requests and Responses on the REST layer, you need to implement `ToXContent` to be able to serialize to and from JSON.

File: src/main/java/org/opensearch/security/tools/SecurityAdmin.java (hunk 13 above)
Comment (nibix) [reply src/main/java/org/opensearch/security/tools/SecurityAdmin.java:1415]: Formatting was a bit difficult for us as different files seem to use different formattings. Do you have a standard formatting policy?

File: src/test/java/org/opensearch/security/SecurityAdminTests.java (hunk 14 above)
Comment (rs-eliatra) [reply src/test/java/org/opensearch/security/SecurityAdminTests.java]: done

File: src/main/java/org/opensearch/security/tools/SecurityAdmin.java (hunk 11 above)
Comment (rs-eliatra) [reply src/main/java/org/opensearch/security/tools/SecurityAdmin.java:278]: https://github.com/opensearch-project/security/pull/1638#discussion_r814084702

File: src/main/java/org/opensearch/security/tools/SecurityAdmin.java (hunk 13 above)
Comment (cliu123) [reply src/main/java/org/opensearch/security/tools/SecurityAdmin.java:1415]: I see. This format is good then. It is consistent with the format in this particular file.

File: src/main/java/org/opensearch/security/action/configupdate/ConfigUpdateResponse.java [hunk 15]
```diff
@@ -58,4 +60,16 @@ public ConfigUpdateResponse(final ClusterName clusterName, List<ConfigUpdateNode
     public void writeNodesTo(final StreamOutput out, List<ConfigUpdateNodeResponse> nodes) throws IOException {
         out.writeList(nodes);
     }
+
+	@Override
+	public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+		builder.startObject("configupdate_response");
```
[the remark this answers is not in the document]
Comment (rs-eliatra) [reply src/main/java/org/opensearch/security/action/configupdate/ConfigUpdateResponse.java:66]: `@Override` annotation is used to inform that it is implementation of interface

From JavaDoc:
```
Indicates that a method declaration is intended to override a method declaration in a supertype. If a method is annotated with this annotation type compilers are required to generate an error message unless at least one of the following conditions hold:
* The method does override or implement a method declared in a supertype.
* The method has a signature that is override-equivalent to that of any public method declared in Object.
```

File: src/main/java/org/opensearch/security/action/configupdate/ConfigUpdateNodeResponse.java [hunk 16]
```diff
@@ -78,4 +80,14 @@ public void writeTo(StreamOutput out) throws IOException {
...
+        builder.startObject();
+        builder.field("updated_config_types", updatedConfigTypes);
+        builder.field("updated_config_size", updatedConfigTypes == null ? 0: updatedConfigTypes.length);
```
[the remark this answers is not in the document]
Comment (rs-eliatra) [reply src/main/java/org/opensearch/security/action/configupdate/ConfigUpdateNodeResponse.java:88]: I believe that it improves readability. 
Imagine that there are hundreds of nodes and someone requires to get only number of nodes. (I believe that there is a lof of usecases for that )

File: src/main/java/org/opensearch/security/action/configupdate/ConfigUpdateResponse.java (hunk 15 above)
[the remark this answers is not in the document]
Comment (peternied) [reply src/main/java/org/opensearch/security/action/configupdate/ConfigUpdateResponse.java:66]: I mention it as it this class extends BaseNodesResponse which was were my confusion was from.  You are correct that the base class doesn't implement the interface so no super call is needed.

**Comment by cliu123** [comment]
@rs-eliatra @nibix 1.3 branch has been created to hold all changes for 1.3 branch. This PR will be included in 2.0 branch, so `main` is the right target branch.

File: src/main/java/org/opensearch/security/tools/SecurityAdmin.java (hunk 10 above)
Comment (cliu123) [reply src/main/java/org/opensearch/security/tools/SecurityAdmin.java:216]: Should the ports at line#212 and line#417 and line#419 be changed?
```
options.addOption(Option.builder("p").longOpt("port").hasArg().argName("port").desc("OpenSearch transport port (default: 9300)").build());
```
```
        if(port < 9300) {
            System.out.println("WARNING: Seems you want connect to the OpenSearch HTTP port."+System.lineSeparator()
                             + "         securityadmin connects on the transport port which is normally 9300.");
        }
```

File: src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+        Assert.assertEquals(HttpStatus.SC_BAD_REQUEST, rh.executePutRequest("_plugins/_security/configupdate", "").getStatusCode());
+        Assert.assertEquals(HttpStatus.SC_OK, rh.executePutRequest("_plugins/_security/configupdate?config_types=roles", "").getStatusCode());
+
+        Assert.assertEquals(HttpStatus.SC_OK, (res = rh.executePutRequest("_plugins/_security/configupdate?config_types=unknown_xxx", "",
+                encodeBasicHeader("nagilum", "nagilum"))).getStatusCode());
+        assertContains(res, "*\"successful\":0*failed_node_exception*");
+
+    }
+
+}
```
Comment (cliu123) [inline src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java:109]: Please add an empty line at EOF

File: src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.test.SingleClusterTest;
+import org.opensearch.security.test.helper.file.FileHelper;
+import org.opensearch.security.test.helper.rest.RestHelper;
+
+public class SecurityAdminIEndpointsTests extends SingleClusterTest {
+
+    @Test public void testNoSSL() throws Exception {
```
Comment (cliu123) [inline src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java:18]: Please put `@Test` on a separate line.

File: src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+                rh.executePutRequest("_plugins/_security/configupdate?config_types=roles", "{}", encodeBasicHeader("nagilum", "nagilum"))
+                        .getStatusCode());
+        Assert.assertEquals(HttpStatus.SC_UNAUTHORIZED, rh.executePutRequest("_plugins/_security/configupdate", "").getStatusCode());
+        Assert.assertEquals(HttpStatus.SC_FORBIDDEN,
+                rh.executePutRequest("_plugins/_security/configupdate?config_types=xxx", "", encodeBasicHeader("nagilum", "nagilum"))
+                        .getStatusCode());
+        Assert.assertEquals(HttpStatus.SC_FORBIDDEN, rh.executeGetRequest("_plugins/_security/whoami").getStatusCode());
+    }
+
+    @Test public void testEndpoints() throws Exception {
```
Comment (cliu123) [inline src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java:33]: Please put `@Test` on a separate line.

File: src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+        final RestHelper rh = nonSslRestHelper();
+
+        Assert.assertEquals(HttpStatus.SC_FORBIDDEN,
+                rh.executePutRequest("_plugins/_security/configupdate?config_types=roles", "{}", encodeBasicHeader("nagilum", "nagilum"))
+                        .getStatusCode());
+        Assert.assertEquals(HttpStatus.SC_UNAUTHORIZED, rh.executePutRequest("_plugins/_security/configupdate", "").getStatusCode());
+        Assert.assertEquals(HttpStatus.SC_FORBIDDEN,
+                rh.executePutRequest("_plugins/_security/configupdate?config_types=xxx", "", encodeBasicHeader("nagilum", "nagilum"))
+                        .getStatusCode());
+        Assert.assertEquals(HttpStatus.SC_FORBIDDEN, rh.executeGetRequest("_plugins/_security/whoami").getStatusCode());
```
Comment (cliu123) [inline src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java:30]: Why this is forbidden with NoSSL?

**Review by cliu123 — DISMISSED** [review]

File: src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java (hunk 18 above)
Comment (cliu123) [inline src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java:18]: Please put @Test on a separate line.

File: src/test/java/org/opensearch/security/SecurityAdminTests.java [hunk 21]
```diff
@@ -88,10 +167,9 @@ public void testSecurityAdminV6Update() throws Exception {
         int returnCode  = SecurityAdmin.execute(argsAsList.toArray(new String[0]));
         Assert.assertNotEquals(0, returnCode);
         
-        RestHelper rh = nonSslRestHelper();
-        HttpResponse res;
-        
-        Assert.assertEquals(HttpStatus.SC_SERVICE_UNAVAILABLE, (res = rh.executeGetRequest("_opendistro/_security/health?pretty")).getStatusCode());
+        RestHelper rh = restHelper();
```
[the remark this answers is not in the document]
Comment (nibix) [reply src/test/java/org/opensearch/security/SecurityAdminTests.java:170]: This test now needs enabled TLS encryption on the REST layer. Thus, no plain-text connections are possible any more.

File: src/main/java/org/opensearch/security/tools/SecurityAdmin.java (hunk 10 above)
Comment (rs-eliatra) [reply src/main/java/org/opensearch/security/tools/SecurityAdmin.java:216]: done

File: src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java (hunk 17 above)
Comment (rs-eliatra) [reply src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java]: done

File: src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java (hunk 19 above)
Comment (rs-eliatra) [reply src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java]: done

File: src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java (hunk 18 above)
Comment (cliu123) [reply src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java]: Good point!

**Review by peternied — DISMISSED** [review]

File: src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java (hunk 18 above)
Comment (peternied) [reply src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java]: I've created a PR that enables enforcement but disables all rules, https://github.com/opensearch-project/security/compare/main...peternied:enforce-checkstyle?expand=1, if we turned these rules on we would run into 35,974 issues to resolve 😮 

### `./gradlew checkstyleMain`
```
  ...
  Checkstyle files with violations: 266
  Checkstyle violations by severity: [error:18588]
```

### `./gradlew checkstyleTest`
```
  ...
  Checkstyle files with violations: 181
  Checkstyle violations by severity: [error:17386]
```

We can enable rules on a piecemeal basis for specific rules or files

Comment (peternied) [reply src/test/java/org/opensearch/security/SecurityAdminIEndpointsTests.java]: Created https://github.com/opensearch-project/security/issues/1659

File: src/main/java/org/opensearch/security/tools/SecurityAdmin.java [hunk 22]
```diff
@@ -421,9 +414,9 @@ public static int execute(final String[] args) throws Exception {
         }
         
         
-        if(port < 9300) {
+        if(port < 9200) {
             System.out.println("WARNING: Seems you want connect to the OpenSearch HTTP port."+System.lineSeparator()
-                             + "         securityadmin connects on the transport port which is normally 9300.");
+                             + "         securityadmin connects on the transport port which is normally 9200.");
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/tools/SecurityAdmin.java:419]: Is this line#417 - line#419 still needed?
`transport port which is normally 9200` - This does not make sense to me.

Comment (rs-eliatra) [reply src/main/java/org/opensearch/security/tools/SecurityAdmin.java:419]: Yes, you are right, I did not get fully the initial message

**Review by cliu123 — APPROVED** [review]

**Review by peternied — APPROVED** [review]

**Comment by cliu123** [comment]
Removing the tag `backport 1.3` because `TransportClient` is removed from OpenSearch 2.0.

**Comment by peternied** [comment]
@cliu123 Is there a reason we do not want this change in 1.3?  Seems like it will more than make the pencils down date

**Comment by cliu123** [comment]
> @cliu123 Is there a reason we do not want this change in 1.3? Seems like it will more than make the pencils down date

This change is not necessary for 1.3.0 since `TransportClient` still exists in 1.3.0. This PR is a part of removing `TransportClient` effort. There will be more PRs to be done, for example #1660. 1.3.0 code freeze is being targeted for next week, I think it will be safer to only merge things that have been completely settled down to 1.3.0 : ). But removing `TransportClient` is still not done yet.
